gui/kitchen-info.lua

```diff
@@ -0,0 +1,433 @@
+local function findTrueTypeSeriesInLine(x1, x2, y, length)
+    local seriesStartX = x1
+    local seriesLength = 1  -- assumes first tile is TrueType
+    for x = x1 + 1, x2 do
+        if (dfhack.screen.readTile(x, y) == nil) then  -- series started or continues
+            if (seriesLength == 0) then  -- series started
+                seriesStartX = x
+            end
+            seriesLength = seriesLength + 1
+        elseif (seriesLength == length) then  -- series of expected length ended
+            break
+        else  -- series of other length ended, reset counter for next series
+            seriesLength = 0
+        end
+    end
+    if (seriesLength == length) then  -- recheck the length of the series at the end, in case the loop ended normally, not because of break
+        return seriesStartX, seriesStartX + length - 1
+    end
+    return -1, -1
+end
+
+local function findStringInLine(x1, x2, y, str)
+    local length = #str
+    local firstByte = str:byte(1)
+    local xEnd = x2 - (length - 1)
+    for x = x1, xEnd do
+        local tile = dfhack.screen.readTile(x, y)
+
+        if (tile == nil) then  -- TrueType text detected
+            return findTrueTypeSeriesInLine(x, x2, y, length)
+        elseif (tile.ch == firstByte) then
+            local matches = true
+            for i = 2, length do
+                tile = dfhack.screen.readTile(x + i - 1, y)
+
+                if (not tile) or (tile.ch ~= str:byte(i)) then
+                    matches = false
+                    break
+                end
+            end
+            if (matches) then
+                return x, x + length - 1
+            end
+        end
+    end
+    return -1, -1
+end
+
+local function drawNumberColumn(x, kitchen, firstVisibleIndex, numDisplayedItems)
+    dfhack.screen.paintString({ fg = COLOR_WHITE }, x, 4, "Number")
+
+    local p = gui.Painter.new_xy(x, 6, x + 10 - 1, 6 + numDisplayedItems - 1)
+    local count = kitchen.count[kitchen.page]
+
+    for i = 0, numDisplayedItems - 1 do
+        local index = firstVisibleIndex + i
+
+        p:seek(0, i)
+        local color = (index == kitchen.cursor) and COLOR_YELLOW or COLOR_BROWN
+        p:pen(color):string(tostring(count[index]))
+    end
+end
+
+local function drawPermissionsColumn(x, kitchen, firstVisibleIndex, numDisplayedItems)
+    dfhack.screen.paintString({ fg = COLOR_WHITE }, x, 4, "Permissions")
+
+    local p = gui.Painter.new_xy(x, 6, x + string.len("Permissions") - 1, 6 + numDisplayedItems - 1)
+    local possible  = kitchen.possible[kitchen.page]
+    local forbidden = kitchen.forbidden[kitchen.page]
+
+    for i = 0, numDisplayedItems - 1 do
+        local index = firstVisibleIndex + i
+
+        p:seek(1, i)
+        if (possible[index].Cook) then
+            local color = forbidden[index].Cook and COLOR_LIGHTRED or COLOR_LIGHTBLUE
```

Comment:
Would it be possible to detect if `tweak kitchen-prefs-color` is enabled so we could display the corresponding color here?